c_glib/gandiva-glib/node.cpp

```diff
@@ -395,42 +434,1438 @@ ggandiva_function_node_get_parameters(GGandivaFunctionNode *node)
+} GGandivaBinaryLiteralNodePrivate;
+
+enum {
+  PROP_BINARY_VALUE = 1
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GGandivaBinaryLiteralNode,
+                           ggandiva_binary_literal_node,
+                           GGANDIVA_TYPE_LITERAL_NODE)
+
+#define GGANDIVA_BINARY_LITERAL_NODE_GET_PRIVATE(object)               \
+  static_cast<GGandivaBinaryLiteralNodePrivate *>(                     \
+    ggandiva_binary_literal_node_get_instance_private(                 \
+      GGANDIVA_BINARY_LITERAL_NODE(object)))
+
+static void
+ggandiva_binary_literal_node_finalize(GObject *object)
+{
+  auto priv = GGANDIVA_BINARY_LITERAL_NODE_GET_PRIVATE(object);
+
+  g_free(priv->value);
+
+  G_OBJECT_CLASS(ggandiva_binary_literal_node_parent_class)->finalize(object);
+}
+
+static void
+ggandiva_binary_literal_node_set_property(GObject *object,
+                                          guint prop_id,
+                                          const GValue *value,
+                                          GParamSpec *pspec)
+{
+  auto priv = GGANDIVA_BINARY_LITERAL_NODE_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_BINARY_VALUE:
+    priv->value = g_value_dup_string(value);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+ggandiva_binary_literal_node_get_property(GObject *object,
+                                          guint prop_id,
+                                          GValue *value,
+                                          GParamSpec *pspec)
+{
+  auto priv = GGANDIVA_BINARY_LITERAL_NODE_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_BINARY_VALUE:
+    g_value_set_string(value, priv->value);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+ggandiva_binary_literal_node_init(GGandivaBinaryLiteralNode *binary_literal_node)
+{
+}
+
+static void
+ggandiva_binary_literal_node_class_init(GGandivaBinaryLiteralNodeClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->finalize     = ggandiva_binary_literal_node_finalize;
+  gobject_class->set_property = ggandiva_binary_literal_node_set_property;
+  gobject_class->get_property = ggandiva_binary_literal_node_get_property;
+
+  GParamSpec *spec;
+  spec = g_param_spec_string("value",
+                             "Value",
+                             "The value of the binary literal",
+                             nullptr,
+                             static_cast<GParamFlags>(G_PARAM_READWRITE |
+                                                      G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_BINARY_VALUE, spec);
+}
+
+/**
+ * ggandiva_binary_literal_node_new:
+ * @value: The value of the binary literal.
+ *
+ * Returns: A newly created #GGandivaBinaryLiteralNode.
+ *
+ * Since: 0.12.0
+ */
+GGandivaBinaryLiteralNode *
+ggandiva_binary_literal_node_new(const gchar *value)
```

Comment:
We need to use `const guint8 *value, gsize size)` for binary data.
Because binary data may contain '`\0`'.

Reply:
I see. I'll try to address them.

Reply:
I've addressed it.